src/overlays/actors/ovl_En_Karebaba/z_en_karebaba.c

```diff
@@ -19,57 +37,480 @@ const ActorInit En_Karebaba_InitVars = {
+        value = 10;
+    }
+
+    this->headCollider.dim.radius = sHeadColliderInit.dim.radius + (value * 2);
+    this->actor.shape.rot.x = 0xC000 - (value * 0x100);
+    this->actor.shape.rot.y += value * 0x2C0;
+    this->actor.posRot.pos.y = (Math_Sins(this->actor.shape.rot.x) * -60.0f) + this->actor.initPosRot.pos.y;
+
+    thing = Math_Coss(this->actor.shape.rot.x) * 60.0f;
+
+    this->actor.posRot.pos.x = (Math_Sins(this->actor.shape.rot.y) * thing) + this->actor.initPosRot.pos.x;
+    this->actor.posRot.pos.z = (Math_Coss(this->actor.shape.rot.y) * thing) + this->actor.initPosRot.pos.z;
+
+    if (this->bodyCollider.base.acFlags & 2) {
+        EnKarebaba_SetupDying(this);
+        func_80032C7C(globalCtx, &this->actor);
+    } else if (this->actor.params == 0) {
+        EnKarebaba_SetupUpright(this);
+    }
+}
+
+void EnKarebaba_Dying(EnKarebaba* this, GlobalContext* globalCtx) {
+    static Vec3f sVecZero1 = { 0.0f, 0.0f, 0.0f };
+
+    s32 i;
+    Vec3f position;
+    Vec3f rotation;
+
+    Math_ApproxF(&this->actor.speedXZ, 0.0f, 0.1f);
+
+    if (this->actor.params == 0) {
+        Math_ApproxUpdateScaledS(&this->actor.shape.rot.x, 0x4800, 0x71C);
+        EffectSsHahen_SpawnBurst(globalCtx, &this->actor.posRot.pos, 3.0f, 0, 12, 5, 1, -1, 10, 0);
+
+        if (this->actor.scale.x > 0.005f && ((this->actor.bgCheckFlags & 2) || (this->actor.bgCheckFlags & 8))) {
+            this->actor.scale.x = this->actor.scale.y = this->actor.scale.z = 0.0f;
+            this->actor.speedXZ = 0.0f;
+            this->actor.flags &= ~5;
+            EffectSsHahen_SpawnBurst(globalCtx, &this->actor.posRot.pos, 3.0f, 0, 12, 5, 15, -1, 10, 0);
+        }
+
+        if (this->actor.bgCheckFlags & 2) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_DODO_M_GND);
+            this->actor.params = 1;
+        }
+    } else if (this->actor.params == 1) {
+        Math_Vec3f_Copy(&position, &this->actor.posRot.pos);
+        rotation.z = Math_Sins(this->actor.shape.rot.x) * 20.0f;
+        rotation.x = -20.0f * Math_Coss(this->actor.shape.rot.x) * Math_Sins(this->actor.shape.rot.y);
+        rotation.y = -20.0f * Math_Coss(this->actor.shape.rot.x) * Math_Coss(this->actor.shape.rot.y);
+
+        for (i = 0; i < 4; i++) {
+            func_800286CC(globalCtx, &position, &sVecZero1, &sVecZero1, 500, 50);
+            position.x += rotation.x;
+            position.y += rotation.z;
+            position.z += rotation.y;
+        }
+
+        func_800286CC(globalCtx, &this->actor.initPosRot.pos, &sVecZero1, &sVecZero1, 500, 100);
+        EnKarebaba_SetupDeadItemDrop(this, globalCtx);
+    }
+}
+
+void EnKarebaba_DeadItemDrop(EnKarebaba* this, GlobalContext* globalCtx) {
+    if (this->actor.params != 0) {
+        this->actor.params--;
+    }
+
+    if (Actor_HasParent(&this->actor, globalCtx) || this->actor.params == 0) {
+        EnKarebaba_SetupDead(this);
+        return;
+    }
+
+    func_8002F554(&this->actor, globalCtx, GI_STICKS_1);
```

Comment:
i would personally do an else instead of a return, but you could argue changing actions warrants an early return. so i leave that up to you